### PR TITLE
Apply guidelines to global variables & modules

### PR DESCRIPTION
**Pull request by MadByteDE**
**NOTE**
**I suggest to merge/resolve #221 first before considering my PR.**

## Overview
This PR applies our style guide to global variables and modules across the project (except for the lib directory of course).

Plus some extra changes:
- Fixes a small bug that causes the terrain end to be visible sometimes after starting a new game
- changes the local `sock` in main.lua to `socket`. Before my changes the global `sock` (sock.lua lib) and the local `sock` (socket lib) where written exactly the same which would likely have led to some confusion sooner or later.
- Removes a duplicate line of `strCurrentScreen = Fun.CurrentScreenName()` in main.lua
- Removes very little whitespace around the local sock definition in main.lua

### CODE REVIEW

File: source/createobjects.lua [hunk 1]
```diff
@@ -2,7 +2,7 @@ local createobjects = {}
 
 
 function createobjects.CreateObject(intType, intXValue)
--- creates a base and appends it to the garrObjects table
```
Comment (togfoxy) [inline source/createobjects.lua:5]: GROUND looks like a constant (as per style guide) but that is a good driver to move away from globals.

File: source/functions.lua [hunk 2]
```diff
@@ -198,33 +198,33 @@ function functions.HandleSockets(dt)
 					msg = {}
 					msg.name = "ConnectionAccepted"
 				else
-					garrLanders[2] = {}			--! super big flaw: this hardcodes garrLanders[2]
-					garrLanders[2].x = incoming.x
-					garrLanders[2].y = incoming.y
-					garrLanders[2].angle = incoming.angle
-					garrLanders[2].name = incoming.name
+					LANDERS[2] = {}			--! super big flaw: this hardcodes LANDERS[2]
```
Comment (togfoxy) [inline source/functions.lua:201]: Is this old code? I thought the hard coding of [2] was removed? I'll double check.

**Review by togfoxy — COMMENTED** [review]
I'll play test this later and make sure nothing got slipped. Thanks.

**Comment by togfoxy** [comment]
There are two trivial conflicts here.

I'll resolve later tonight then merge and then raise a PR for the range finder issue.

Then I'll release v0.11.

**Comment by Philbywhizz** [comment]
I'm getting crashes since this merge

```
Error: objects/terrain.lua:141: attempt to index global 'enum' (a nil value)
stack traceback:
	[string "boot.lua"]:777: in function '__index'
	objects/terrain.lua:141: in function 'generate'
	objects/terrain.lua:96: in function 'init'
	functions.lua:241: in function 'ResetGame'
	main.lua:209: in function 'load'
	[string "boot.lua"]:586: in function <[string "boot.lua"]:585>
	[C]: in function 'xpcall'
	[string "boot.lua"]:793: in function <[string "boot.lua"]:780>
	[C]: in function 'xpcall'

```

**Comment by togfoxy** [comment]
The PR's are done out of order. I look forward to us not moving so fast and thing settling down.

enum should be Enum which is fixed in the RADARFIX pull request. If you look at the RADARFIX PR then you'll see this offending line of code is completely deleted.

I'll merge RADARFIX when it is ready then return to this PR. You think that will work?

**Comment by MadByteDE** [comment]
Sure try it.

>  I look forward to us not moving so fast and thing settling down.
the problems we have all are pretty minor like wrong cased names and duplicate code. We just changed the entire code structure, so for what I know I think it went pretty smooth.

Another topic:
@togfoxy: you merged the code and resolved the conflicts but while resolving it seems like you forgot to remove the redundant code snippet and ended up with both variants merged to one:

```lua
	local sock = require 'socket'	-- socket is native to LOVE but needs a REQUIRE
	HOST_IP_ADDRESS = sock.dns.toip(sock.dns.gethostname())

	local socket = require 'socket'	-- socket is native to LOVE but needs a REQUIRE
	HOST_IP_ADDRESS = socket.dns.toip(socket.dns.gethostname())
	socket = nil
```

should be:
```lua
	local socket = require 'socket'	-- socket is native to LOVE but needs a REQUIRE
	HOST_IP_ADDRESS = socket.dns.toip(socket.dns.gethostname())
	socket = nil
```
I'd suggest togfoxy fixes this to make sure nobody else change it back again.

**Pull request merged**